internal/pkg/manifest/workload.go

```diff
@@ -74,6 +76,31 @@ type Workload struct {
 	Type *string `yaml:"type"` // must be one of the supported manifest types.
 }
 
+// OverrideRule holds the manifest overriding rule for CloudFormation template.
+type OverrideRule struct {
+	TaskDefOverrideRules []TaskDefinitionOverrideRule `yaml:"taskdef_override"`
+}
+
+// TaskDefinitionOverrideRule holds the overriding rule for ECS Task Definition.
+type TaskDefinitionOverrideRule string
```

Comment:
It's a bit odd to read that an OverrideRule is composed of a TaskDefOverrideRule. Usually the more specific type embeds the generic type name like: 
```go
type Bicycle struct { }
type RoadBike struct {
  Bicycle
}

```

If we go with the proposal below then we can do:
```go
type OverrideRule struct {
   Path  string     `yaml:"path"`
   Value *yaml.Node `yaml:"value"`
}

type BackendServiceConfig struct {
  ... // our existing fields

  TaskDefOverrides []OverrideRule `yaml:"taskdef_overrides"`
}
```

Reply:
Can we also make the field plural so that it matches our other array fields like `success_codes` or `allowed_source_ips`? 🙏